miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt

```diff
@@ -54,10 +58,34 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
 
     /**
      * Receive the files from the HostApp.
-     * @param files The array of Uri to be invoked by a filePathCallback after successfully retrieved
-     * by [Activity.onActivityResult] in the HostApp.
+     * @param intent The data after successfully retrieved by [Activity.onActivityResult] in the HostApp.
      */
-    fun onReceivedFiles(files: Array<Uri>) {
-        callback?.onReceiveValue(files)
+    @Suppress("OptionalWhenBraces", "LongMethod")
+    fun onReceivedFiles(intent: Intent) {
+        val data = intent.data
+        val clipData = intent.clipData
+        when {
+            data != null -> {
+                callback?.onReceiveValue((arrayOf(data)))
+            }
+            clipData != null -> {
+                val uriList = mutableListOf<Uri>()
+                for (i in 0 until clipData.itemCount) {
+                    uriList.add(clipData.getItemAt(i).uri)
+                }
+
+                callback?.onReceiveValue((uriList.toTypedArray()))
+            }
+            else -> {
+                callback?.onReceiveValue(null)
+            }
+        }
+    }
+
+    /**
+     * Can be used when HostApp wants to reset the file choosing operation.
+     */
+    fun reset() {
```

Comment:
Nice solution!

I think maybe the name could be specific on where this is used. How about something like `onCancel` or `cancelReceivingFiles`?

Reply:
Yeah, I have renamed to `onCancel`.